mp/src/game/client/neo/ui/neo_hud_ammo.cpp

```diff
@@ -137,169 +123,170 @@ void CNEOHud_Ammo::DrawAmmo() const
 	surface()->DrawSetTextPos((text_xpos + xpos) - fontWidth, text_ypos + ypos);
 	surface()->DrawPrintText(unicodeWepName, textLen);
 
 	if(dynamic_cast<C_WeaponGhost*> (activeWep))
-	{
 		return;
-	}
 
 	const int maxClip = activeWep->GetMaxClip1();
-	if (maxClip != 0 && !activeWep->IsMeleeWeapon())
-	{
-		const int ammoCount = activeWep->m_iPrimaryAmmoCount;
-		const int numClips = ceil(abs((float)ammoCount / activeWep->GetMaxClip1())); // abs because grenades return negative values (???) // casting division to float in case we have a half-empty mag, rounding up to show the half mag as one more mag
-		const auto isSupa = dynamic_cast<CWeaponSupa7*>(activeWep);
+	if (maxClip == 0 || activeWep->IsMeleeWeapon())
+		return;
+
+	const int ammoCount = activeWep->m_iPrimaryAmmoCount;
+	const int numClips = ceil(abs((float)ammoCount / activeWep->GetMaxClip1())); // abs because grenades return negative values (???) // casting division to float in case we have a half-empty mag, rounding up to show the half mag as one more mag
+	const auto isSupa = dynamic_cast<CWeaponSupa7*>(activeWep);
 		
-		if (activeWep->UsesClipsForAmmo1()) {
-			const int maxLen = 5;
-			char clipsText[maxLen]{ '\0' };
-			if(isSupa)
-			{
-				snprintf(clipsText, 10, "%d+%d", ammoCount, activeWep->m_iSecondaryAmmoCount.Get());
-			} else
-			{
-				snprintf(clipsText, 10, "%d", numClips);
-			}
-
-			textLen = V_strlen(clipsText);
-			wchar_t unicodeClipsText[maxLen]{ L'\0' };
-			g_pVGuiLocalize->ConvertANSIToUnicode(clipsText, unicodeClipsText, sizeof(unicodeClipsText));
-
-			int clipsTextWidth, clipsTextHeight;
-			surface()->GetTextSize(m_hTextFont, unicodeClipsText, clipsTextWidth, clipsTextHeight);
-			surface()->DrawSetTextFont(m_hTextFont);
-
-			surface()->GetTextSize(m_hTextFont, unicodeClipsText, fontWidth, fontHeight);
-			surface()->DrawSetTextPos(digit2_xpos + xpos - fontWidth, digit2_ypos + ypos);
-			surface()->DrawPrintText(unicodeClipsText, textLen);
+	if (activeWep->UsesClipsForAmmo1()) {
+		const int maxLen = 5;
+		char clipsText[maxLen]{ '\0' };
+		if(isSupa)
+		{
+			V_sprintf_safe(clipsText, "%d+%d", ammoCount, activeWep->m_iSecondaryAmmoCount.Get());
+		} else
+		{
+			V_sprintf_safe(clipsText, "%d", numClips);
 		}
 
-		const auto neoWep = dynamic_cast<C_NEOBaseCombatWeapon*> (activeWep);
+		textLen = V_strlen(clipsText);
+		wchar_t unicodeClipsText[maxLen]{ L'\0' };
+		g_pVGuiLocalize->ConvertANSIToUnicode(clipsText, unicodeClipsText, sizeof(unicodeClipsText));
+
+		int clipsTextWidth, clipsTextHeight;
+		surface()->GetTextSize(m_hTextFont, unicodeClipsText, clipsTextWidth, clipsTextHeight);
+		surface()->DrawSetTextFont(m_hTextFont);
+
+		surface()->GetTextSize(m_hTextFont, unicodeClipsText, fontWidth, fontHeight);
+		surface()->DrawSetTextPos(digit2_xpos + xpos - fontWidth, digit2_ypos + ypos);
+		surface()->DrawPrintText(unicodeClipsText, textLen);
+	}
+
+	const auto neoWep = dynamic_cast<C_NEOBaseCombatWeapon*> (activeWep);
```

Comment:
This cast is redundant now - can use `activeWep` (which is `C_NEOBaseCombatWeapon*`) instead of declaring `neoWep`.